lib/iris/_merge.py

```diff
@@ -1218,8 +1224,10 @@ def merge(self, unique=True):
             if all_have_data:
                 # All inputs were concrete, so turn the result back into a
                 # normal array.
+                dtype = self._cube_signature.data_type
                 merged_data = as_concrete_data(merged_data,
-                                               nans_replacement=ma.masked)
+                                               nans_replacement=ma.masked,
+                                               result_dtype=dtype)
```

Comment:
This was an interesting original omission that only surfaced through the act of `DataManager` integration, as an existing test was failing ... there is just so not enough test coverage!